### ISSUE TITLE
quer params parse boolean seem has a little problem

### ISSUE THREAD

**Issue by xihuanhh** [pleerock/routing-controllers#124]
when i send  "true" or "false" ,it works fine;
then i send 0 or 1 , first  query params all be passed as string, to be "0" or "1", 
so it  all be parse as  true , not expect  false and true.

i wonder the source code **/util/ParamHandler.ts**    

        switch (formatName.toLowerCase()) {
            case "number":
                return +value;

            case "string":
                return value;

            case "boolean":
                if (value === "true") {
                    return true;
                    
                } else if (value === "false") {
                    return false;
                }
                return !!value;

            default:
                const isObjectFormat = format instanceof Function || formatName.toLowerCase() === "object";
                if (value && (param.parseJson || isObjectFormat))
                    value = this.parseValue(value, param);
        }
        return value;
    }

**Comment by pleerock** [comment]
Im sorry its hard for me to understand your english, can you please describe your issue once again? Do you have problem when you send "0" and its true? Or what?

**Comment by xihuanhh** [comment]
haha, i am so sorry for my pool english, my means is that  when i want parse 0(**number**) or 1(**number**)  to **boolean** in query params during an http request , actual they will be converted to "0"(**string**) and "1" (**string**) first, because "0" or "1" is not empty string, they all will be parsed as **true** which is not  my expect . @pleerock

**Comment by xihuanhh** [comment]
haha, i am so sorry for my pool english, my means is that  when i want parse 0(**number**) or 1(**number**)  to **boolean** in query params during an http request ; actually,  they will be converted to "0"(**string**) and "1" (**string**) first, because "0" or "1" is not empty string in javascript, they all will be parsed as **true**  that which is not correctly in  my expect  .     @pleerock

**Comment by xihuanhh** [comment]
for example. 
```javascript 
 @Get('/test')  
 async test(@QueryParam('isFlag',{required:true}) isFlag:boolean){
         assert(isFlag === true); // true
         assert(isFlag === false);// false
   }
```

then  visit  http://localhost:3000/test?isFlag=0 like this , the variable **isFlag** will always be **true** 
@pleerock

**Comment by NoNameProvided** [comment]
I have experienced the same, while tried to send boolean param as `0` or `1`. I just ended up sending `true` and `false` as string. What @xihuanhh would like to discuss is an implementation like this:

```ts
switch (formatName.toLowerCase()) {
        case "number":
            return +value;

        case "string":
            return value;

        case "boolean":
            if (value === "true" || value === "1") {
                return true;
                
            } else if (value === "false" || value === "0") {
                return false;
            }
            return !!value;

        default:
            const isObjectFormat = format instanceof Function || formatName.toLowerCase() === "object";
            if (value && (param.parseJson || isObjectFormat))
                value = this.parseValue(value, param);
    }
    return value;
}
``` 

So he would like to see `0` to be casted to `false` and `1` to `true`. When the type of the param is `boolean`

**Comment by xihuanhh** [comment]
@NoNameProvided  yeah, that's just I want, thanks a lot!

**Comment by pleerock** [comment]
okay, done released in `0.7.0-alpha.14`
